docs/Development/ReleaseNotes.md

```diff
@@ -51,14 +51,16 @@ To check which release of VIC you are running:
 
     1. Fixes Julian day for the first timestep in the dmy struct for the CESM driver.
 
-3. Speed up NetCDF operations in the image/CESM drivers ([GH#684](https://github.com/UW-Hydro/VIC/pull/684))
```

Comment:
I'm confused by the numbering and the edits on previous pull requests. Please fix